### PR TITLE
[Sprint 4] Feature confirm request off logic

### PR DESCRIPTION
**Pull request by tranminhtri911**
- https://dev.framgia.com/redmine/issues/122999
- Feature confirm request off logic
- Effect : None

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffPresenter.java [hunk 1]
```diff
@@ -9,8 +21,17 @@
...
 
     private ConfirmRequestOffContract.ViewModel mViewModel;
+    private RequestRepository mRequestRepository;
+    private UserRepository mUserRepository;
+    private BaseSchedulerProvider mSchedulerProvider;
+    private CompositeDisposable mCompositeDisposable;
 
-    ConfirmRequestOffPresenter() {
+    ConfirmRequestOffPresenter(UserRepository userRepository, RequestRepository requestRepository,
+            BaseSchedulerProvider schedulerProviderA) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffPresenter.java]: ``schedulerProviderA`` -> ``schedulerProvider``

**Comment by NguyenTienVinh** [comment]
Approved

File: app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffContract.java [hunk 2]
```diff
@@ -11,11 +14,21 @@
...
+        void onGetUserSuccess(User user);
+
+        void onGetUserError(BaseException exception);
     }
 
     /**
      * Presenter.
      */
     interface Presenter extends BasePresenter<ViewModel> {
+        void onCreateForm(RequestOff requestOff);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffContract.java]: ``onCreateForm`` -> ``createFormRequestOff``
Presenter contain action

File: app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffModule.java [hunk 3]
```diff
@@ -2,10 +2,22 @@
...
+import com.framgia.wsm.data.source.remote.UserRemoteDataSource;
 import com.framgia.wsm.utils.dagger.ActivityScope;
+import com.framgia.wsm.utils.navigator.Navigator;
+import com.framgia.wsm.utils.rx.BaseSchedulerProvider;
+import com.framgia.wsm.widget.dialog.DialogManager;
+import com.framgia.wsm.widget.dialog.DialogManagerImpl;
 import dagger.Module;
 import dagger.Provides;
 
+import static com.framgia.wsm.utils.Constant.EXTRA_REQUEST_OFF;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffModule.java]: pls remove this import

File: app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffModule.java [hunk 4]
```diff
@@ -22,13 +34,42 @@ public ConfirmRequestOffModule(@NonNull Activity activity) {
     @ActivityScope
     @Provides
     public ConfirmRequestOffContract.ViewModel provideViewModel(
-            ConfirmRequestOffContract.Presenter presenter) {
-        return new ConfirmRequestOffViewModel(presenter);
+            ConfirmRequestOffContract.Presenter presenter, Navigator navigator,
+            DialogManager dialogManager) {
+        RequestOff requestOff = mActivity.getIntent().getExtras().getParcelable(EXTRA_REQUEST_OFF);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/confirmrequestoff/ConfirmRequestOffModule.java]: ``EXTRA_REQUEST_OFF`` -> ``Constant.EXTRA_REQUEST_OFF``

File: app/src/main/res/layout/activity_confirm_request_off.xml [hunk 5]
```diff
@@ -130,8 +134,10 @@
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/dp_10"
                         android:orientation="horizontal"
+                        android:visibility="@{viewModel.visiableProjectName ? View.VISIBLE : View.GONE}"
                         >
                         <ImageView
+                            android:id="@+id/imageView2"
```
Comment (daolq3012) [inline app/src/main/res/layout/activity_confirm_request_off.xml]: pls remove this id

**Pull request merged**